code/modules/ai/ai_holder_targeting.dm

```diff
@@ -220,10 +236,11 @@
 
 // Sets a few vars so mobs that threaten will react faster to an attacker or someone who attacked them before.
 /datum/ai_holder/proc/on_attacked(atom/movable/AM)
-	last_conflict_time = world.time
 	if(isliving(AM))
 		var/mob/living/L = AM
-		attackers |= L.name
+		if(!(L in attackers))
```

Comment:
Not sure this is the correct check since ``attackers`` contains mob names and not mob instances.